MobileWallet/Backup/ICloudBackup.swift

```diff
@@ -0,0 +1,342 @@
+ Running on macOS 10.15
+ 
+ Copyright 2019 The Tari Project
+ 
+ Redistribution and use in source and binary forms, with or
+ without modification, are permitted provided that the
+ following conditions are met:
+ 
+ 1. Redistributions of source code must retain the above copyright notice,
+ this list of conditions and the following disclaimer.
+ 
+ 2. Redistributions in binary form must reproduce the above
+ copyright notice, this list of conditions and the following disclaimer in the
+ documentation and/or other materials provided with the distribution.
+ 
+ 3. Neither the name of the copyright holder nor the names of
+ its contributors may be used to endorse or promote products
+ derived from this software without specific prior written permission.
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
+ CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+ OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import Foundation
+
+enum ICloudBackupWalletError: Error {
+    case failedToCreateArchive
+    case noAnyBackups
+    case unzipError
+    case dbFileNotFound
+    case iCloudContainerNotFound
+    case unableCreateBackupFolder
+    case privateKeyError
+}
+
+extension ICloudBackupWalletError: LocalizedError {
+    public var errorDescription: String? {
+        switch self {
+        case .failedToCreateArchive:
+            return NSLocalizedString("Failed to create wallet backup archive", comment: "backup archive error description")
+        case .noAnyBackups:
+            return NSLocalizedString("You have not any wallet backup", comment: "'No any wallet backup' error description")
+        case .unzipError:
+            return NSLocalizedString("Unable to unarchive wallet from iCloud backup", comment: "unarchive wallet error description")
+        case .dbFileNotFound:
+            return NSLocalizedString("Unable to create wallet backup. File not found", comment: "sqlite file not found error description")
+        case .iCloudContainerNotFound:
+            return NSLocalizedString("Unable to create wallet backup. iCloud container not found", comment: "iCloud container not found error description")
+        case .unableCreateBackupFolder:
+            return NSLocalizedString("Unable to create backup folder", comment: "Unable to create backup folder error descroption")
+        case .privateKeyError:
```

Comment:
From a user's perspective these 3 are the same: `failedToCreateArchive`, `dbFileNotFound`, `unableToCreateBackupFolder`. They're just backup errors, and it won't help the user to give detailed information here, they just can't do anything about it if either of these happens. Yet, `iCloudContainerNotFound` could be fixed by the user by logging into their Apply ID, or enabling Cloud Drive on the device - so you can have split `iCloudContainerNotFound` into two.